pkg/build/controller/strategy/util.go

```diff
@@ -139,6 +142,22 @@ func setupSourceSecrets(pod *kapi.Pod, sourceSecret *kapi.LocalObjectReference)
 	}...)
 }
 
+// setupSecretBuildSources mounts the secrets referenced by the SecretBuildSource
+// into a builder container. It also sets an environment variable that contains
+// a name of the secret and the destination directory.
+func setupSecretBuildSources(pod *kapi.Pod, secrets []buildapi.SecretBuildSource) {
+	for _, s := range secrets {
+		mountSecretVolume(pod, s.Secret.Name, filepath.Join(SecretBuildSourceBaseMountPath, s.Secret.Name), "build")
```

Comment:
similarly i'd be worried here about allowing users to append arbitrary secret name values as a path.  what if my secret is named "../../.."?  could something bad happen here?